lib/rbac.ex

```diff
@@ -37,4 +37,117 @@ defmodule RBAC do
   def transform_role_list_to_string(roles) do
     [Map.delete(roles, :__meta__)] |> transform_role_list_to_string()
   end
+
+  @doc """
+  `get_approles/2` fetches the roles for the app
+  """
+  def get_approles(auth_url, client_id) do
+    url = "#{auth_url}/approles/#{client_id}"
+    HTTPoison.start()
+
+    HTTPoison.get(url)
+    |> parse_body_response()
+  end
+
+  @doc """
+  `parse_body_response/1` parses the response
+  so your app can use the resulting JSON (list of roles).
+  """
+  @spec parse_body_response({atom, String.t()}) :: String.t()
+  def parse_body_response({:error, err}), do: {:error, err}
```

Comment:
Should these be private functions defined with `defp`? I don't think they will need to be called by any external modules

Reply:
@yeah, not much danger of them being called externally, but you're 100% correct they should be `defp` as we aren't unit testing them directly. 👍

Reply:
updated to `defp` ✅